webapp/src/hooks/general.ts

```diff
@@ -345,14 +347,75 @@ export function usePost(postId: string) {
 }
 
 export function useRun(runId: string, teamId?: string, channelId?: string) {
-    return useThing(runId, fetchPlaybookRun, (state) => {
-        const runsByTeam = myPlaybookRunsByTeam(state);
-        if (teamId && channelId) {
-            // use efficient path
-            return runsByTeam[teamId]?.[channelId];
+    return useThing(runId, fetchPlaybookRun, getRun(runId, teamId, channelId));
+}
+
+export enum FetchState {
+    idle = 'idle',
+    pending = 'pending',
+    done = 'done',
+    error = 'error',
+}
+
+export type FetchMetadata = {
+    state: FetchState;
+    error: ClientError | null;
+}
+
+/**
+ *
+ * @param id The id of the resource to be fetched
+ * @param fetch The function used to make the fetch
+ * @param deps Additional deps that might be needed to trigger again the fetch func
+ * @returns array tuple with the data in the first position and the fetchState in the second
+ */
+export function useFetch<T>(
+    id: string,
+    fetch: (id: string) => Promise<T>,
```

Comment:
Could we rename this so we don't shadow the global `fetch` function? The suggested name is just a proposal, any other thing is good for me:
```suggestion
    fetchFunction: (id: string) => Promise<T>,
```

Reply:
good catch